srcpkgs/volta/template

```diff
@@ -0,0 +1,77 @@
+# Template file for 'volta'
+pkgname=volta
+version=0.7.1
+revision=2
+archs="x86_64 i686"
+build_helper="rust"
+hostmakedepends="cargo pkg-config"
+makedepends="libressl-devel"
+short_desc="JavaScript tool version sync helper"
+maintainer="Alex Lohr <alex.lohr@logmein.com>"
+license="BSD-2-Clause"
+homepage="https://volta.sh/"
+distfiles="https://github.com/volta-cli/${pkgname}/archive/v${version}.tar.gz"
+checksum=e53a07e167bb64103f36901423f5a377a2ea89ecfdd7a1343e69d659f99f9c1b
+
+post_patch() {
+	cat >> shell/unix/install.sh <<_EOF
+createDir() {
+	if [ ! -d "\$1" ]; then
+		mkdir "\$1"
+	fi
+}
+
+createSymlink() {
+	if [ ! -e "\$2" ]; then
+		ln -s "\$1" "\$2"
+	fi
+}
+
+installProfile() {
+	if [ -e "\$1" ]; then
+	  isInstalled=\$(grep "\$2" "\$1")
+		if [ -z "\$isInstalled" ]; then
+			echo "" >> "\$1"
+			echo "export VOLTA_HOME=\"\$HOME/.volta\"" >> "\$1"
+			echo "export PATH=\"\$HOME/.volta/bin:\\\$PATH\"" >> "\$1"
+			echo "source \$2" >> "\$1"
+		fi
+	fi
+}
+
+# Create directories
+createDir "\$HOME/.volta"
+createDir "\$HOME/.volta/bin"
+
+# Create symlinks
+createSymlink "/usr/bin/volta" "\$HOME/.volta/volta"
+createSymlink "/usr/bin/volta-migrate" "\$HOME/.volta/volta-migrate"
+createSymlink "/usr/bin/volta-shim" "\$HOME/.volta/volta-shim"
+
+# Install profile
+installProfile "\$HOME/.bashrc" "/etc/volta/load.bash"
+installProfile "\$HOME/.cshrc" "/etc/volta/load.sh"
+installProfile "\$HOME/.config/fish/config.fish" "/etc/volta/load.fish"
+installProfile "\$HOME/.zshrc" "/etc/volta/load.sh"
+_EOF
+}
+
+do_build() {
+	cargo build --release --target ${RUST_TARGET}
+}
+
+do_install() {
+	vlicense LICENSE
+
+	vbin target/${RUST_TARGET}/release/volta
+	vbin target/${RUST_TARGET}/release/volta-shim
+	vbin target/${RUST_TARGET}/release/volta-migrate
+
+	vmkdir etc/volta 755
+	vinstall shell/unix/load.bash 755 etc/volta
+	vinstall shell/unix/load.sh 755 etc/volta
+	vinstall shell/unix/load.fish 755 etc/volta
+	vinstall shell/unix/install.sh 755 etc/volta
+
+	echo "To install volta locally, use etc/volta/install.sh"
```

Comment:
This is going to just cast  a message into the void.  Build logs are rarely read, and you shouldn't clutter them with meaningless messages.

Reply:
Any idea how to do that better?

Reply:
The install script is now no longer mandatory, it just makes setup easier. The rest is already documented in the volta documentation.